inst/make_temp_dir__.m

```diff
@@ -0,0 +1,50 @@
+%% Copyright (C) 2022 Alex Vong
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <https://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @deftypefun {@var{path} =} make_temp_dir__ (@var{tmpdir}, @var{prefix})
+%% Try to create temporary directory in temporary directory @var{tmpdir} with
+%% prefix @var{prefix} in the most secure and portable way possible.
+%%
+%% This function is not really intended for end users.
+%%
+%% @end deftypefun
+
+
+function path = make_temp_dir__ (tmpdir, prefix)
+  cmd = {'import tempfile'
+         '(tmpdir, prefix) = _ins'
+         'return tempfile.mkdtemp(dir=tmpdir, prefix=prefix)'};
+  path = pycall_sympy__ (cmd, tmpdir, prefix);
```

Comment:
not a huge fan of having temp dir creation depend on Python...!

Is this really not do-able on Octave?

Reply:
AFAIK octave has no function to create temporary directory.

If java is enabled, we can just use java. For *nix, we can use the shell command `mktemp -d`... So basically we are left with the case of octave without java running on windows. Do you know if all octave builds on windows include java? If it's the case, problem solved!

Reply:
> Do you know if all octave builds on windows include java?

Java isn't included in the installer of Octave for Windows. Octave just tries to detect if a compatible JRE "happens" to be installed that it can use.
So, you shouldn't rely on being able to use Java.

Reply:
I see. Since cygwin / msys2 is part of octave in windows (as indicated in #1182), I think we can use the `mktemp` program.